### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by patrickisgreene** [little-dude/xi-tui#65]
xi-term need's to be refactored in order to continue adding components, Id like to change the directory structure to make this easier. I don't plan on changing any code with this change just organize the modules and files.

I think there the best way to do this(for now at least) is to create two sub-modules, 'core' & 'widgets'.

The core module will hold files like tui.rs and terminal.rs as well as provide a place to add 'TUI' code and other core functionalities.

The widgets module will hold all the on screen display items. eg. Editor Views, Status bar, and Command prompt, as well as any future widgets we decide to implement.

So TLDR; I'd like this to be the project structure at least for the time being:
```
xi-term/
   ├──core
          ├──tui.rs
          ├──terminal.rs
          └──mod.rs
   ├──widgets
           |──view
           |    └────same as current
           └──mod.rs
   ├──main.rs
```

**Comment by little-dude** [comment]
Sounds like a good idea to me!
